### ISSUE TITLE
compute server filesystem: directory caching issue

### ISSUE THREAD

**Issue by williamstein** [sagemathinc/cocalc#7546]
1. I uploaded a file directly onto the compute server (so in file explorer the Server dropdown was set to the compute server)
2. But directly on the compute server the file wouldn't read even though it appeared via ls.
3. I clicked the sync button and then everything was fine.

<img width="1127" alt="image" src="https://github.com/sagemathinc/cocalc/assets/1276278/d5c0ee65-c3e8-445f-91d7-0a6cbced1c43">

Basically there is a hopefully easy to reproduce issue involving caching of file metadata.  

WORKAROUND: If a file is there but empty or can't be read, click the "Sync" button. 

In a lot of cases, manually clicking "sync" is necessary if you want the filesystem to sync immediately.  That's by design.  However, in this case -- uploading files via the UI -- definitely clicking sync should not be required.
